pylint/checkers/deprecated.py

```diff
@@ -26,12 +26,16 @@
 
 
 class DeprecatedMixin(BaseChecker):
-    """A mixin implementing logic for checking deprecated symbols.
 
-    A class implementing mixin must define "deprecated-method" Message.
-    """
+    DEPRECATED_IMPORT_MSGS: dict[str, MessageDefinitionTuple] = {
+        "W0402": (
+            "Deprecated module %r",
+            "deprecated-module",
+            "A module marked as deprecated is imported.",
+        ),
+    }
 
-    msgs: dict[str, MessageDefinitionTuple] = {
+    DEPRECATED_MSGS: dict[str, MessageDefinitionTuple] = {
```

Comment:
This is a question how DeprecatedMixin is supposed to be used:
* should external users reuse DeprecatedMixin?
  * if yes should they use MSGS which are used also in STDLIB checker (W15XX) and import checker (W0402)? If W15XX and W0402 are reused also for external users the **renaming is confusing then** because it is not connected to STDLIB only.
  * if no we should somehow parametrize the mixin allowing users to specify which message should be raised
* if external users should not reuse DeprecatedMixin we can do the renaming.

Reply:
> should external users reuse DeprecatedMixin? 

I'm wondering why we do not have a configuration for deprecations checks ? Isn't it possible to have user modifiable ``deprecated_functions``, ``deprecated_classes`` list options ?

Reply:
> I'm wondering why we do not have a configuration for deprecations checks ? Isn't it possible to have user modifiable deprecated_functions, deprecated_classes list options ?

We can provide to the user these options. Should it go with this PR?

In any case I think some users will need subclassing `DeprecatedMixin`. By subclassing mixin user can check the deprecated function also based on version - e.g. django plugin can check deprecated django functions/classes based on django version. This cannot be done by simple list of functions/modules in configuration file.

Reply:
> We can provide to the user these options. Should it go with this PR?

Let's first handle this PR.

> By subclassing mixin user can check the deprecated function also based on version - e.g. django plugin can check deprecated django functions/classes based on django version.

Ha right, it make sense.